src/repositories/versionable/VersionableRepository.ts

```diff
@@ -40,6 +40,34 @@ export default class VersionableRepository<D extends mongoose.Document, M extend
         return this.model.findOne(data);
     }
 
+    public async getall(skipDefined: number, limitDefined: number, sort: boolean) {
+        try {
+            if ( sort ) {
+                    const fetchData = await this.model.find( { deletedAt : undefined})
+                    .skip(skipDefined)
+                    .limit(limitDefined)
+                    .sort({name: 1, email: 1});
+                    const count = await this.model.find( {deletedAt: undefined})
+                    .countDocuments();
+                    const arr = [fetchData, count];
+                    return arr;
+            }
+            else {
+                    const fetchData = await this.model.find({deletedAt: undefined})
+                    .skip(skipDefined)
+                    .limit(limitDefined)
+                    .sort({createdAt: -1});
+                    const count = await this.model.find({deletedAt: undefined})
+                    .countDocuments();
+                    const arr = [fetchData, count];
+                    return arr;
+            }
+        }
```

Comment:
Implement this functionality in UserRepository class, Since versionableRepository is a generic class and can be utilized by any model. So it's not the right place to put model-specific code here.